llvm/lib/CodeGen/MachineSink.cpp

```diff
@@ -784,7 +805,18 @@ bool MachineSinking::runOnMachineFunction(MachineFunction &MF) {
       // of a def-use chain, if there is any.
       // TODO: Sort the candidates using a cost-model.
       unsigned i = 0;
+
```

Comment:
This is sinking 50 instructions. My suggestions would be to update FindCycleSinkCandidates and sink only those  candidates which have low latency. There is already precedence to look at target. Ideally we don't want sink high latency instructions into the loops.